src/build.ts

```diff
@@ -137,9 +137,19 @@ async function _build (nitro: Nitro, rollupConfig: RollupConfig) {
   await _snapshot(nitro)
 
   nitro.logger.start('Building server...')
-  const build = await rollup.rollup(rollupConfig).catch((error) => {
-    nitro.logger.error('Rollup error: ' + error.message)
-    throw error
+  const build = await rollup.rollup(rollupConfig).catch((_error) => {
+    try {
+      for (const error of ('errors' in _error ? _error.errors : [_error])) {
+        const id = error.id || _error.id
+        let path = isAbsolute(id) ? relative(process.cwd(), id) : id
+        const location = error.loc || error.location
+        if (location) {
+          path += `:${location.line}:${location.column}`
+        }
+        nitro.logger.error(`Rollup error while processing \`${path}\`` + '\n' + '\n' + (error.text || error.frame))
```

Comment:
Maybe we can refactor this in a function utility `showRollupError` (could be even same file) and reuse for https://github.com/unjs/nitro/blob/0aa77a9ed8e125beab281e5b630990d38096e66f/src/build.ts#L207